### ISSUE TITLE
Google Drive sync isn't working in either direction

### ISSUE THREAD

**Issue by tedric42** [donmor/Tiddloid#18]
I just got a new Motorola One 5G and am trying to get Tiddloid working with a wiki file hosted in my Google Drive. I can link and edit the wiki no problem, but when I try to save it quickly results in an error "Upload failed / An error occurred". I am fully signed into Drive on this device.

I then made some changes to the wiki via my laptop which uses a browser extension to autosave changes. When I returned to my phone, Tiddloid claimed the wiki file I'd loaded before no longer existed. I re-imported the wiki file but it did not reflect the new changes (even though the Drive thumbnail clearly shows some new Tiddlers I created on my laptop). After clearing the cache of both apps, Tiddloid claimed the file wasn't a wiki file. I then closed and reopened both apps, which allowed me to finally import the wiki file with the new changes, but when I tried to make edits with Tiddloid again I got the same "Upload failed" error.

Any ideas? At this point I'm thinking maybe there's some conflict between the Tiddly Drive extension I use on my laptop and Tiddloid? I thought Google Drive hosting would be the simplest way to keep a private wiki that I could use & edit from multiple devices but maybe I need to find other tools instead.

**Comment by donmor** [comment]
Does OneDrive work? I can't access to Google's services, so I tested with OneDrive and everything is fine, except that Tiddloid crashed on deleting the source file in the app. I'll be working on it soon.

**Comment by donmor** [comment]
In your case, it seems GDrive won't sync the new file if cached exists, and won't sync the changes in the cache.
Tiddloid uses Android's standard SAF APIs, and it won't know which kind of malfunction it is outside. It just checks the previously provided file's validity and prompts you if file is invalid.

**Comment by tedric42** [comment]
Aha! It looks like all I had to do was mark the file for "offline access" in Drive. Your note about cache gave me the idea. Right after doing that, I was able to make changes locally on my phone and have it sync successfully right away, and confirm them seconds later on the computer -- and the same in reverse. I'll let you know if it gives me any more trouble but hopefully it will stay this smooth forever! This might be something to note in documentation.
